core/src/main/java/org/kiwix/kiwixmobile/core/main/CompatFindActionModeCallback.kt

```diff
@@ -112,36 +106,34 @@ class CompatFindActionModeCallback internal constructor(context: Context) :
   // If true, find the next match further down in the document.
   // If false, find the previous match, up in the document.
   private fun findNext(next: Boolean) {
-    if (webView == null) {
-      throw AssertionError("No WebView for CompatFindActionModeCallback::findNext")
-    }
-    webView?.findNext(next)
+    requireNotNull(webView) {
+      "No WebView for CompatFindActionModeCallback::findNext"
+    }.findNext(next)
   }
 
   // Highlight all the instances of the string from mEditText in mWebView.
   fun findAll() {
-    if (webView == null) {
-      throw AssertionError("No WebView for CompatFindActionModeCallback::findAll")
+    requireNotNull(webView) {
+      "No WebView for CompatFindActionModeCallback::findAll"
     }
-    val find: CharSequence? = editText.text
-    if (find == null || find.isEmpty()) {
-      webView?.clearMatches()
-      webView?.findAllAsync("")
-    } else {
-      webView?.findAllAsync("$find")
+    val textToFind: CharSequence? = editText.text
+    if (textToFind?.isNotEmpty() == true) {
+      webView?.findAllAsync("$textToFind")
 
       // Enable word highlighting with reflection
       try {
-        for (ms in WebView::class.java.declaredMethods) {
-          if (ms.name == "setFindIsUp") {
-            ms.isAccessible = true
-            ms.invoke(webView, true)
-            break
+        WebView::class.java.declaredMethods
+          .firstOrNull { it.name == "setFindIsUp" }
+          ?.apply {
```

Comment:
`let` is probably more idiomatic,

![image](https://user-images.githubusercontent.com/3358282/96858843-af8f1700-1458-11eb-9454-0d8baa44cf24.png)

this [article](https://proandroiddev.com/kotlin-scope-and-extension-functions-b35925ec96b7) I largely agreed with

Reply:
Run from run!
9f00d4e